src/bridging/subscription/change.c

```diff
@@ -0,0 +1,757 @@
+				// get bridge interface and set its address
+				error = rtnl_link_get_kernel(socket, 0, br_component_name, &bridge);
+				if (error) {
+					SRPLG_LOG_ERR(PLUGIN_NAME, "rtnl_link_get_kernel() failed (%d): %s", error, nl_geterror(error));
+					goto error_out;
+				}
+
+				// convert to libnl mac address format
+				mac_address_ly_to_nl((char *) node_value);
+				error = nl_addr_parse(node_value, AF_LLC, &address);
+				if (error < 0) {
+					SRPLG_LOG_ERR(PLUGIN_NAME, "nl_addr_parse() failed (%d): %s", error, nl_geterror(error));
+					goto error_out;
+				}
+
+				// configure link for changes
+				bridge_changes = rtnl_link_bridge_alloc();
+				rtnl_link_set_name(bridge_changes, br_component_name);
+				rtnl_link_set_addr(bridge_changes, address);
+
+				error = rtnl_link_change(socket, bridge, bridge_changes, 0);
+				if (error < 0) {
+					SRPLG_LOG_ERR(PLUGIN_NAME, "rtnl_link_change() failed (%d): %s", error, nl_geterror(error));
+					goto error_out;
+				}
+				break;
+			case SR_OP_DELETED:
+				// address cannot be deleted
+			case SR_OP_MOVED:
+				break;
+		}
+	}
+
+error_out:
+	if (bridge) {
+		rtnl_link_put(bridge);
+	}
+	if (bridge_changes) {
+		rtnl_link_put(bridge_changes);
+	}
+	if (address) {
+		nl_addr_put(address);
+	}
+	return error;
+}
+
+int modify_filtering_database(struct nl_sock *socket, char change_path[PATH_MAX], const char *node_name, const char *node_value, sr_change_oper_t operation)
+{
+	int error = 0;
+	struct rtnl_link *bridge = NULL;
+	sr_xpath_ctx_t xpath_ctx = {0};
+	char *bridge_component_name = NULL;
+	uint32_t seconds = 0;
+	int bridge_link_idx = 0;
+
+	if (strcmp(node_name, "aging-time") == 0) {
+		switch (operation) {
+			case SR_OP_CREATED:
+			case SR_OP_MODIFIED:
+				bridge_component_name = sr_xpath_key_value(change_path, "component", "name", &xpath_ctx);
+				error = rtnl_link_get_kernel(socket, 0, bridge_component_name, &bridge);
+				if (error) {
+					SRPLG_LOG_ERR(PLUGIN_NAME, "rtnl_link_get_kernel() error: (%d) %s", error, nl_geterror(error));
+					goto error_out;
+				}
+				bridge_link_idx = rtnl_link_get_ifindex(bridge);
+
+				seconds = (uint32_t) strtoul(node_value, NULL, 10);
+				error = bridge_set_ageing_time(socket, bridge_link_idx, seconds);
+				if (error != 0) {
+					SRPLG_LOG_ERR(PLUGIN_NAME, "bridge_set_ageing_time() failed (%d)", error);
+					goto error_out;
+				}
+				break;
+			case SR_OP_DELETED:
+				break;
+			case SR_OP_MOVED:
+				break;
+		}
+	} else if (strstr(change_path, "vlan-registration-entry")) {
+		error = modify_port_vlan_entries(socket, change_path, node_name, node_value, operation);
+		if (error) {
+			SRPLG_LOG_ERR(PLUGIN_NAME, "modify_port_vlan_entries() failed (%d): %s", error);
+			goto error_out;
+		}
+	} else if (strstr(change_path, "filtering-entry")) {
+		error = modify_filtering_entries(socket, change_path, node_name, node_value, operation);
+		if (error) {
+			SRPLG_LOG_ERR(PLUGIN_NAME, "modify_filtering_entries() failed (%d): %s", error);
```

Comment:
The `%s` argument will read random data from the stack, since a 2nd argument isn't provided